lib/timers.js

```diff
@@ -272,8 +309,36 @@ function setImmediate(callback, arg1, arg2, arg3) {
   return new Immediate(callback, args);
 }
 
-setImmediate[customPromisify] = function(value) {
-  return new Promise((resolve) => new Immediate(resolve, [value]));
+setImmediate[customPromisify] = function(value, options = {}) {
+  if (options == null || typeof options !== 'object') {
+    return Promise.reject(
+      new ERR_INVALID_ARG_TYPE(
+        'options',
+        'Object',
+        options));
+  }
+  const { signal } = options;
+  if (signal !== undefined &&
+    (signal === null ||
+     typeof signal !== 'object' ||
+     !('aborted' in signal))) {
+    return Promise.reject(
+      new ERR_INVALID_ARG_TYPE(
+        'options.signal',
+        'AbortSignal',
+        signal));
+  }
+  if (signal && signal.aborted)
+    return Promise.reject(lazyDOMException('AbortError'));
+  return new Promise((resolve, reject) => {
+    const immediate = new Immediate(resolve, [value]);
+    if (signal) {
+      signal.onabort = () => {
```

Comment:
doesn't this clobber anything else that uses `onabort`?

Reply:
Yep... the question there would be: is that ok or should we add a listener instead?

Reply:
I assume we should add a listener, an AbortController should be able to control multiple operations right?

Reply:
Updated to `addEventListener`